content/docs/engine-dev/architecture/prefabs/_index.md

```diff
@@ -0,0 +1,645 @@
+| Component       | A grouping of properties that helps define the behavior of the entity. |
+| Prefab Template | An in-memory representation of the prefab. It represents the current state of prefab loaded in memory. This includes the file loaded from disk and any changes you applied on top of it. |
+| Prefab Instance | The actual prefab game object that is instantiated by using the prefab template. A single prefab template can be used to instantiate any number of prefab instances. |
+| Link            | An object used to express the connection between a source template and a target template. It contains a list of patches. |
+| Patch           | Represents a JSON patch that is applied on top of the nested prefab instance to distinguish it from other nested instances that are instantiated from the same template (the source template). |
+| DOM             | Stands for Document Object Model. It is a language agnostic logical model that represents any object data in a tree-like hierarchical structure where each node in the tree holds some relevant information about the data. In prefab system, DOM values are represented by [RapidJSON](https://rapidjson.org/) objects. Prefab DOM types are defined in [`PrefabDomTypes`](https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabDomTypes.h). |
+
+
+
+
+## Overview
+<!-- <a name="overview"> -->
+
+To get an understanding of what prefabs are, we first need to understand what entities are. Entities are the foundational game objects that encapsulate logic required to build game worlds. An entity is made up of components that define its properties. More information about entities and components can be found in [Guide to Component Development](https://www.o3de.org/docs/user-guide/programming/components/). 
+
+Prefabs serve the purpose of easily building and modifying large and complex worlds for games. Creating large worlds often requires duplicating a lot of entities in the world. While this can be done by simply creating a bunch of entities and duplicating them in the editor, modifying them becomes a pain point since you would have to go into each of those entities and make the same changes. You would also have to come up with custom tracking mechanisms to identify which entities to modify in which areas of the world. In addition, you may also want some entities to retain their old properties and not to change along with other entities. All of these problems are resolved by prefabs.
+
+Simply put, a prefab is a collection of entities and nested prefabs. The further sections will explain how this collection of entities and nested prefabs are used to solve the above problems and many more.
+
+
+### Prefab file format
+<!-- <a name="prefab-file-format"> -->
+
+Prefab files use JSON format. A prefab file has the following fields.
+
+<dl>
+  <dt>Container Entity</dt>
```

Comment:
I added these tags for better organization on GitHub wiki.

Need docs' eyes on this. I am not sure if we can use these tags in docs. Same for `<details>` below.

Reply:
Seems like https://www.o3de.org/docs/contributing/to-docs/style-guide/format/#definition-lists is what we need. Changing...